### ISSUE TITLE
wireless problems: how to revert?

### ISSUE THREAD

**Issue by orcaomar** [sabhiram/raspberry-wifi-conf#9]
so, this is what happens to me:
- everything works, but when i connect to the AP, i don't get any webpage (at the address with port 88)
- when i go to localhost:88 on the pi itself, i get the webpage, but don't see any APs
- now my WiFi is hosed -- i can't seem to reconfigure back into normal wifi mode.

question: how do i undo everything that this program has done? is there a flag i can use? maybe some uninstall instructions are in order in case the whole installation fails.

**Comment by sabhiram** [comment]
Oh snap, that sounds terrible!

What happens when you reboot the Pi after the Wifi being hosed? Worst case, you can update the /etc/network/interfaces file to look like the correct version for your wifi connection at home / work.

Take a look at "./assets/etc/network/interfaces.wifi.template" which should be written to "/etc/network/interfaces".

Hope this helps!

**Comment by orcaomar** [comment]
hey sorry for my really delayed reply. first, your fix does work, in terms of bringing back wifi. i'll switch back to your wifi later today and post what happens on reboot. again, sorry for the delay -- i was travelling and not near my pi

**Comment by sabhiram** [comment]
No worries at all! Glad that resolves it for you - When I find some time I will get to fixing this in a cleaner way. Could be as simple as a helper script that just does a "reset" to whatever known good "old" state.

**Comment by orcaomar** [comment]
yes, that sounds like a good idea. ok, now when i reboot i get this error:

[FAIL] Starting ISC DHCP server: dhcpd[....] check syslog for diagnostics. ... failed!
  failed!
My IP address is 192.168.1.158
[FAIL] startpar: service(s) returned failure: isc-dhcp-server ... failed!

this is after i've set things back to the way they were. so perhaps restoring to the way they were includes stopping this dhcp server from running at startup? can you point me to where this server is getting inserted in your code?

**Comment by sabhiram** [comment]
Check out this file: ./assets/etc/default/isc-dhcp-server.template

And look at how it is used here: https://github.com/sabhiram/raspberry-wifi-conf/blob/master/app/wifi_manager.js#L194

**Comment by sabhiram** [comment]
Any update @orcaomar ?

**Comment by hakankaraduman** [comment]
I have the same problem, I reinstalled and executed your app, I can reach the server on the pi itself and try to connect my ssid, it says ap is closed and wifi is enabled, I restarted the pi, changed the /etc/network/interfaces to wifi template but I still have the 'no wireless interfaces found' text. Is there a way to recover this without reinstalling the raspbian os

**Comment by sabhiram** [comment]
Hah @hakankaraduman unfortunately I did not spend too much time in the reverse direction since I usually would just replace all the relevant service / config files using this method.

**Comment by vicwomg** [comment]
I just experienced this as well. Lots of hours rebooting, modifying files, and enabling/disabling services trying to get my wifi back up and running again. I think that at the very least the old conf files should be backed up before they are overwritten, and there should be a troubleshooting warning about this in the documentation.

To anyone else having issues here are the steps I took to reverse the changes and recover wifi on my pi, not sure if everything was necessary but made sure to write everything down:

1. Edit /etc/dnsmasq.conf , set to a blank file
2. Edit /etc/hostapd/hostapd.conf , set to a blank file
3. Edit /etc/dhcpcd.conf , comment out or remove these lines at the end: 
```
interface wlan0
static ip_address=192.168.88.1/24
nohook wpa_supplicant
```

4. Add the following to the end of your /etc/network/interfaces
```
allow-hotplug wlan0 
iface wlan0 inet manual
wpa-conf /etc/wpa_supplicant/wpa_supplicant.conf
```
5. Edit your /etc/wpa_supplicant/wpa_supplicant.conf to your wifi ap 
```
ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
update_config=1
country=<your 2-letter country code>
network={
  ssid="<your router ssid>"
  psk="<your pw>"
  key_mgmt=WPA-PSK
}
```
6. Re-enable the wpa_supplicant service
`sudo systemctl enable  wpa_supplicant.service`
7. Remove raspberry wifi conf service
`/etc/init.d/raspberry-wifi-conf`
8. reboot
